Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/keysubmission/srs/AnalyticsSrsKeySubmissionStorage.kt

```diff
@@ -0,0 +1,52 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.keysubmission.srs
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import de.rki.coronawarnapp.datadonation.analytics.modules.keysubmission.AnalyticsSrsKeySubmissionDataStore
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AnalyticsSrsKeySubmissionStorage @Inject constructor(
+    @AnalyticsSrsKeySubmissionDataStore private val dataStore: DataStore<Preferences>
+) {
+    private val dataStoreFlow = dataStore.data
```

Comment:
could be lazy

Reply:
yes, it could. why?

Reply:
Performance 😁

Reply:
😄 I know what `lazy` is for, I meant why here specifically. You refactored the datastores non of them is using lazy when accessing the flow.

Reply:
Yes, you are right, didn't realised beforehand, but this would be one additional refactoring which could make the datastore approach a tiny bit more efficient.

Reply:
It depends, it not necessarily true that using `lazy` will improve the performance. it could make it worse. see https://stackoverflow.com/questions/65543747/why-does-using-by-lazy-make-my-kotlin-code-slower